packages/theme/src/theme.ts

```diff
@@ -51,12 +48,9 @@ export interface ThemeData {
 }
 
 export class Theme extends HTMLElement {
-    private static themeFragments: FragmentMap = new Map();
-    private static defaultFragments: Set<FragmentName> = new Set([
-        'core',
-        'light',
-        'medium',
-    ]);
+    private hasAdoptedStyles = false;
+    private static themeFragmentsByKind: Map<string, FragmentMap> = new Map();
+    private static defaultFragments: Set<FragmentName> = new Set(['core']);
```

Comment:
With `name` in the fragment now, we might be able to generate these values from `themeFragmentsByKind`... 🤔 this is slightly faster than looking it up every time, and this data is predominately used in the polyfilled version, so it's probably better to have a static list and to not worry too much about it as it will go away over time.